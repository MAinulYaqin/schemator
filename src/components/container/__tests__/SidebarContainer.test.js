import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import ConnectedComponent, { SidebarContainer } from "../SidebarContainer";

describe("StatusbarContainer", () => {
  const setup = propOverrides => {
    const props = {
      page: "workarea",
      toPage: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<SidebarContainer {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = () => {
    const mockStore = configureStore();
    const store = mockStore({
      page: "workarea"
    });

    store.dispatch = jest.fn();

    const wrapper = shallow(<ConnectedComponent store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClickItem' while item was clicked", () => {
    const { wrapper, props } = setup();
    const item = { id: "workarea" };

    wrapper
      .find("Sidebar")
      .props()
      .onClickItem(item);

    expect(props.toPage).toBeCalledWith(item.id);
  });

  it("should maps state to props", () => {
    const { wrapper, store } = setupConnected();
    const { page } = store.getState();

    expect(wrapper.props()).toMatchObject({ page });
  });

  it("should maps dispatch to props", () => {
    const { wrapper, store } = setupConnected();
    const { toPage } = wrapper.props();
    const pageID = "extensions";
    const expected = {
      type: "SET_PAGE",
      payload: pageID
    };

    toPage(pageID);

    expect(store.dispatch).toBeCalledWith(expected);
  });
});
