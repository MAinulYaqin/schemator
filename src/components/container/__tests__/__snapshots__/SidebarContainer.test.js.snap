// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusbarContainer should renders default 1`] = `
<Sidebar
  active="workarea"
  items={
    Array [
      Object {
        "icon": [Function],
        "id": "workarea",
        "tooltip": "Work Area",
      },
      Object {
        "icon": [Function],
        "id": "plugins",
        "tooltip": "Plugins",
      },
    ]
  }
  onClickItem={[Function]}
/>
`;
